src/platform/K32W/BLEManagerImpl.cpp

```diff
@@ -37,7 +58,1415 @@ namespace Internal {
+
+    // Start advertising if needed...
+    if (mServiceMode == ConnectivityManager::kCHIPoBLEServiceMode_Enabled && GetFlag(mFlags, kFlag_AdvertisingEnabled))
+    {
+        // Start/re-start advertising if not already started, or if there is a pending change
+        // to the advertising configuration.
+        if (!GetFlag(mFlags, kFlag_Advertising) || GetFlag(mFlags, kFlag_RestartAdvertising))
+        {
+            err = StartAdvertising();
+            SuccessOrExit(err);
+        }
+    }
+    // Otherwise, stop advertising if it is enabled.
+    else if (GetFlag(mFlags, kFlag_Advertising))
+    {
+        err = StopAdvertising();
+        SuccessOrExit(err);
+        ChipLogProgress(DeviceLayer, "Stopped Advertising");
+    }
+
+exit:
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(DeviceLayer, "Disabling CHIPoBLE service due to error: %s", ErrorStr(err));
+        mServiceMode = ConnectivityManager::kCHIPoBLEServiceMode_Disabled;
+    }
+}
+
+void BLEManagerImpl::DriveBLEState(intptr_t arg)
+{
+    sInstance.DriveBLEState();
+}
+
+/*******************************************************************************
+ * BLE App Task Processing
+ *******************************************************************************/
+void BLEManagerImpl::bleAppTask(void * p_arg)
+{
+    while (1)
+    {
+        xEventGroupWaitBits(bleAppTaskLoopEvent, LOOP_EV_BLE, true, false, portMAX_DELAY);
+
+        PlatformMgr().LockChipStack();
+
+        if (MSG_Pending(&blekw_msg_list))
+        {
+            /* There is message from the BLE tasks to solve */
+            blekw_msg_t * msg = (blekw_msg_t *) MSG_DeQueue(&blekw_msg_list);
+
+            assert(msg != NULL);
+
+            if (msg->type == BLE_KW_MSG_ERROR)
+            {
+                ChipLogProgress(DeviceLayer, "BLE Fatal Error: %d.\n", msg->data.u8);
+            }
+            else if (msg->type == BLE_KW_MSG_CONNECTED)
+            {
+                sInstance.HandleConnectEvent(msg);
+            }
+            else if (msg->type == BLE_KW_MSG_DISCONNECTED)
+            {
+                sInstance.HandleConnectionCloseEvent(msg);
+            }
+            else if (msg->type == BLE_KW_MSG_MTU_CHANGED)
+            {
+                blekw_start_connection_timeout();
+                ChipLogProgress(DeviceLayer, "BLE MTU size has been changed to %d.", msg->data.u16);
+            }
+            else if (msg->type == BLE_KW_MSG_ATT_WRITTEN || msg->type == BLE_KW_MSG_ATT_LONG_WRITTEN ||
+                     msg->type == BLE_KW_MSG_ATT_CCCD_WRITTEN)
+            {
+                sInstance.HandleWriteEvent(msg);
+            }
+            else if (msg->type == BLE_KW_MSG_ATT_READ)
+            {
+                blekw_att_read_data_t * att_rd_data = (blekw_att_read_data_t *) msg->data.data;
+                ChipLogProgress(DeviceLayer, "Attribute read request(d:%d, h:%d) .", att_rd_data->device_id, att_rd_data->handle);
+                /* TODO: */
```

Comment:
add some description to the TODO? what needs to be done?

Reply:
Removed entirely this branch as it's not exercised for this use-case.